### PR TITLE
JavaScript: Make access paths more reusable

### PR DESCRIPTION
**Pull request by max-schaefer**
As discussed, this factors out a few reusable predicates from the access-path computation. The API graphs currently don't use this yet, so there is no particular rush to merge this, I just wanted to put it up for visibility and discussion.

An [evaluation](https://github.com/max-schaefer/dist-compare-reports/blob/master/js/make-access-paths-more-reusable/report.md) (internal link) shows a slight performance degradation. The massive apparent slowdown on gecko for the first run was due to the base SHA timing out on `definitions.ql` and the tip SHA timing out on a later query; on rerun, both timed out on `definitions.ql`.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/GlobalAccessPaths.qll [hunk 1]
```diff
@@ -124,172 +124,243 @@ module AccessPath {
...
+    }
+
+    /**
+     * Holds if `succ` is assigned to the same access path as `pred`, but with `step` appended.
+     *
+     * This predicate is mostly for internal use.
+     */
+    cached
+    predicate rhs2rhs(DataFlow::Node pred, DataFlow::Node succ, string step) {
+      any(DataFlow::PropWrite pw).writes(pred, step, succ)
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/GlobalAccessPaths.qll:328]: This predicate seems to be identical to `ref2rhs`.

Comment (max-schaefer) [reply javascript/ql/src/semmle/javascript/GlobalAccessPaths.qll:328]: Ah, yes; I suppose that isn't really very useful. I'll remove it.

**Comment by max-schaefer** [comment]
Thanks for the review, @asgerf! Comments addressed.

**Comment by max-schaefer** [comment]
Looks like we won't immediately need this, so closing for now.

**Pull request closed without merging**